pkg/gatewayserver/io/mqtt/mqtt.go

```diff
@@ -319,6 +328,13 @@ func (c *connection) deliver(pkt *packet.PublishPacket) {
 			logger.WithError(err).Warn("Failed to unmarshal Tx acknowledgment message")
 			return
 		}
+		if ack.DownlinkMessage == nil {
```

Comment:
So do we or don't we support the downlink message to be passed by the client?

Reply:
The proposed implementation forwards the `DownlinkMessage` as-is if it is provided by the gateway, otherwise it tries to match using the correlation ID of the downlink token. Should we always ignore the `DownlinkMessage` sent by the gateway instead?

Reply:
> Should we always ignore the `DownlinkMessage` sent by the gateway instead?

Yes, and put in the API doc that it's set by the Gateway Server.